server/src/document/DbtTextDocument.ts

```diff
@@ -130,32 +130,33 @@ export class DbtTextDocument {
       TextDocument.update(this.rawDocument, params.contentChanges, params.textDocument.version);
       this.requireCompileOnSave = true;
     } else {
-      const compiledContentChanges = params.contentChanges.map<TextDocumentContentChangeEvent>(c => {
-        if (!TextDocumentContentChangeEvent.isIncremental(c)) {
+      const compiledContentChanges: TextDocumentContentChangeEvent[] = [];
+
+      for (const change of params.contentChanges) {
+        if (!TextDocumentContentChangeEvent.isIncremental(change)) {
           throw new Error('Incremental updates expected');
         }
-        return {
-          text: c.text,
-          range: Range.create(
-            this.convertPosition(this.compiledDocument.getText(), this.rawDocument.getText(), c.range.start),
-            this.convertPosition(this.compiledDocument.getText(), this.rawDocument.getText(), c.range.end),
-          ),
-        };
-      });
+
+        const compiledStart = Diff.convertPositionStraight(this.rawDocument.getText(), this.compiledDocument.getText(), change.range.start);
+        const compiledEnd = Diff.convertPositionStraight(this.rawDocument.getText(), this.compiledDocument.getText(), change.range.end);
+
+        if (compiledStart === undefined || compiledEnd === undefined) {
+          TextDocument.update(this.rawDocument, params.contentChanges, params.textDocument.version);
+          this.requireCompileOnSave = true;
+          return;
+        }
```

Comment:
Could you explain what happens here? Why we can have compiledStart or compiledEnd undefined, and why we set requireCompileOnSave to true?

Reply:
Due to algorithm change it does not return undefined anymore.